src/wordIntelli.py

```diff
@@ -0,0 +1,247 @@
+        hypernyms = []
+        for token in tokens:
+            syns = wn.synsets(token)
+            if len(syns) > 0 and len(syns[0].hypernyms()) > 0:
+                hns = syns[0].hypernyms()[0]
+                hypernyms.append(hns.lemmas()[0].name())
+            else:
+                hypernyms.append(token)
+        return hypernyms # " ".join(hypernyms).strip()
+
+    def get_hyponyms(self, tokens):
+        # tokens = word_tokenize(line)
+        hyponyms = []
+        for token in tokens:
+            syns = wn.synsets(token)
+            if len(syns) > 0 and len(syns[0].hyponyms()) > 0:
+                hns = syns[0].hyponyms()[0]
+                hyponyms.append(hns.lemmas()[0].name())
+            else:
+                hyponyms.append(token)
+        return hyponyms
+
+    def get_meronyms(self, tokens):
+        # tokens = word_tokenize(line)
+        meronyms = []
+        '''Other meronyms available are substance_meronyms, member_meronyms'''
+        for token in tokens:
+            synsets = wn.synsets(token)
+            if len(synsets) > 0 and (len(synsets[0].part_meronyms()) > 0 or len(synsets[0].member_meronyms()) > 0) :
+                if len(synsets[0].part_meronyms()) > 0: # syns[0]
+                    hns = synsets[0].part_meronyms()[0]
+                else:
+                    hns = synsets[0].member_meronyms()[0]
+                meronyms.append(hns.lemmas()[0].name())
+            else:
+                meronyms.append(token)
+        return meronyms
+
+    def get_holonyms(self, tokens):
+        # tokens = word_tokenize(line)
+        holonyms = []
+        '''Other meronyms available are substance_meronyms, member_meronyms'''
+        for token in tokens:
+            synsets = wn.synsets(token)
+            if len(synsets) > 0 and (len(synsets[0].part_holonyms()) > 0 or len(synsets[0].member_holonyms()) > 0) :
+                if len(synsets[0].part_holonyms()) > 0: # syns[0]
+                    hns = synsets[0].part_holonyms()[0]
+                else:
+                    hns = synsets[0].member_holonyms()[0]
+                holonyms.append(hns.lemmas()[0].name())
+            else:
+                holonyms.append(token)
+        return holonyms #" ".join(holonyms).strip()
+
+
+    def get_lemmas(self, tokens, pos_tags):
+        # tags = word_tokenize(pos_tags)
+        # text = word_tokenize(line)
+        wnl = WordNetLemmatizer()
+        lemmas = []
+        for i in range(0, len(tokens)):
+            tag = self.get_wordnet_pos(pos_tags[i])
+            if tag != '':
+                lemmas.append(str(wnl.lemmatize(tokens[i], tag)))
+            else:
+                lemmas.append(str(wnl.lemmatize(tokens[i])))
+        return lemmas # " ".join(lst).strip() # return
+
+    '''Adds JSON object to Solr core'''
+    def add_fields(self, data, core):
+        print("connecting to Solr.....")
+        solr = pysolr.Solr('http://localhost:8983/solr/'+core,always_commit=True)
```

Comment:
Is this connecting to local host port 8983 on the local machine? What happens if this port is being used on that machine?